.idea/Working version/algorithms.js

```diff
@@ -0,0 +1,472 @@
+
+
+                        p23 = a_edges.edgeProb.subset(math.index(j, k));
+                        if (p23) {
+
+
+                            count = 0;
+                            faceFlag = [];
+
+                            if ( p12==1 && p23==1 && p13==1 ){
+                                // all pEdges=1 so pFace=1, no iteration needed
+                                p = 1;
+                                // if 2 edges have pEdge=1 then pFace will be equal to prob of remaining edge
+                            } else if ( p12==1 && p23==1) {
+                                p = p13;
+                                // type = '2 edges'
+                            } else if ( p12==1 && p13==1 ) {
+                                p = p23;
+                                // type = '2 edges'
+                            } else if ( p23==1 && p13==1 ) {
+                                p = p12;
+                                // type = '2 edges'
+                            } else {
+                                // edge probs <1 so must iterate through all permutations
+                                e12 = a_edges.edgeFlag.subset(math.index(i,j)).iEdgeFlag;
+                                e13 = a_edges.edgeFlag.subset(math.index(i,k)).iEdgeFlag;
+                                e23 = a_edges.edgeFlag.subset(math.index(j,k)).iEdgeFlag;
+                                for (ii=0; ii<numPoints; ii++) {
+                                    for (jj=0; jj<numPoints; jj++) {
+                                        for (kk=0; kk<numPoints; kk++) {
+                                            if ( e12[ii][jj] && e13[ii][kk] && e23[jj][kk]) {
+                                                faceFlag.push([ii, jj, kk])
+                                                count++
+                                            }
+                                        }
+                                    }
+                                }
+                                p = count/Math.pow(numPoints,3);
+                                // type = 'computed'
+                            }
+
+                            // if prob is >0 add a new face and proceed to next iteration
+                            if (p) {
+                                tempFaces.push({Pt1: i, Pt2: j, Pt3: k, Pface: p, iFaceFlag: faceFlag.slice()})
+                            }
+
+                        }
+                    }
+                }
+
+
+            }
+        }
+    }
+
+
+    var mapped = tempFaces.map( function (d,i) {
+        return { index: i, value: d.Pface }
+    })
+
+    mapped.sort( function(a, b) {
+        return +(a.value > b.value) || +(a.value === b.value) - 1;
+    })
+
+    ripsFaces = mapped.map( function (d, i) {
+        return tempFaces[d.index]
+    })
+
+    constructCech()
+
+}
+
+
+//construct the Cech complex
+function constructCech() {
+
+
+    cechEdges = JSON.parse(JSON.stringify(ripsEdges));
```

Comment:
There's no need to stringify/parse because ripEdges is already parsed when you loaded the file.